pkg/util/validation.go

```diff
@@ -18,19 +18,33 @@ package util
 
 import (
 	"regexp"
-	"strings"
 )
 
-// IsDNSLabel tests for a string that conforms to the definition of a label in
-// DNS (RFC 1123).
-func IsDNSLabel(value string) bool {
-	return IsDNS1123Label(value)
+const kubeChar string = "[A-Za-z0-9]"
+const extendedKubeChar string = "[-A-Za-z0-9_.]"
+const qualifiedToken string = "(" + kubeChar + extendedKubeChar + "*)?" + kubeChar
+
+const LabelValueFmt string = "((" + kubeChar + extendedKubeChar + "*)?" + kubeChar + ")?"
+
+var labelValueRegexp = regexp.MustCompile("^" + LabelValueFmt + "$")
+
+const LabelValueMaxLength int = 63
+
+func IsValidLabelValue(value string) bool {
+	return (len(value) <= LabelValueMaxLength && labelValueRegexp.MatchString(value))
 }
 
-// IsDNSSubdomain tests for a string that conforms to the definition of a
-// subdomain in DNS (RFC 1123).
-func IsDNSSubdomain(value string) bool {
-	return IsDNS1123Subdomain(value)
+// Annotation values are opaque.
+func IsValidAnnotationValue(value string) bool {
+	return true
+}
+
+const QualifiedNameFmt string = "(" + qualifiedToken + "/)?" + qualifiedToken
+
+var qualifiedNameRegexp = regexp.MustCompile("^" + QualifiedNameFmt + "$")
+
+func IsQualifiedName(value string) bool {
+	return (len(value) <= DNS1123SubdomainMaxLength && qualifiedNameRegexp.MatchString(value))
```

Comment:
You have a problem conflating label and qnames, yet you use DNS's length for qname? :)